src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -302,62 +298,35 @@ public SNIHandle CreateConnectionHandle(object callbackObject, string fullServer
         {
             instanceName = new byte[1];
 
+            DataSource details = DataSource.ParseServerName(fullServerName);
+            if (details == null)
+            {
+                return null;
+            }
+
             SNIHandle sniHandle = null;
-            if (fullServerName.IndexOf(':') == -1)
+
+            if (details.ConnectionProtocol == DataSource.Protocol.None)
             {
                 // default to using tcp if no protocol is provided
-                sniHandle = CreateTcpHandle(fullServerName, timerExpire, callbackObject, parallel, ref spnBuffer, isIntegratedSecurity);
+                sniHandle = CreateTcpHandle(details, timerExpire, callbackObject, parallel, ref spnBuffer, isIntegratedSecurity);
```

Comment:
Why does this need its own if block rather than just having it be a `case DataSource.Protocol.None` in the switch below?